packages/mui-material-next/src/Button/Button.test.js

```diff
@@ -1,263 +1,110 @@
-  });
-
-  it('should render a secondary outlined button', () => {
-    const { getByRole } = render(
-      <Button variant="outlined" color="secondary">
-        Hello World
-      </Button>,
-    );
-    const button = getByRole('button');
-
-    expect(button).to.have.class(classes.root);
-    expect(button).to.have.class(classes.outlined);
-    expect(button).to.have.class(classes.outlinedSecondary);
-    expect(button).not.to.have.class(classes.text);
-    expect(button).not.to.have.class(classes.textSecondary);
-    expect(button).not.to.have.class(classes.contained);
-  });
-
-  it('should render an inherit outlined button', () => {
-    const { getByRole } = render(
-      <Button variant="outlined" color="inherit">
-        Hello World
-      </Button>,
-    );
-    const button = getByRole('button');
-
-    expect(button).to.have.class(classes.root);
-    expect(button).to.have.class(classes.outlined);
-    expect(button).to.have.class(classes.colorInherit);
-    expect(button).not.to.have.class(classes.text);
-    expect(button).not.to.have.class(classes.textSecondary);
-    expect(button).not.to.have.class(classes.contained);
-  });
-
-  it('should render a contained button', () => {
-    const { getByRole } = render(<Button variant="contained">Hello World</Button>);
-    const button = getByRole('button');
-
-    expect(button).to.have.class(classes.root);
-    expect(button).not.to.have.class(classes.text);
-    expect(button).not.to.have.class(classes.textPrimary);
-    expect(button).not.to.have.class(classes.textSecondary);
-    expect(button).to.have.class(classes.contained);
-  });
-
-  it('should render a contained primary button', () => {
-    const { getByRole } = render(
-      <Button variant="contained" color="primary">
-        Hello World
-      </Button>,
-    );
-    const button = getByRole('button');
-
-    expect(button).to.have.class(classes.root);
-    expect(button).not.to.have.class(classes.text);
-    expect(button).to.have.class(classes.contained);
-    expect(button).to.have.class(classes.containedPrimary);
-    expect(button).not.to.have.class(classes.containedSecondary);
-  });
-
-  it('should render a contained secondary button', () => {
-    const { getByRole } = render(
-      <Button variant="contained" color="secondary">
-        Hello World
-      </Button>,
-    );
-    const button = getByRole('button');
-
-    expect(button).to.have.class(classes.root);
-    expect(button).not.to.have.class(classes.text);
-    expect(button).to.have.class(classes.contained);
-    expect(button).not.to.have.class(classes.containedPrimary);
-    expect(button).to.have.class(classes.containedSecondary);
-  });
+    expect(button).to.have.class(classes.text);
+    expect(button).to.have.class(classes.colorSecondary);
+    expect(button).not.to.have.class(classes.colorPrimary);
+  });
+
+  ['filled', 'filledTonal', 'outlined', 'elevated'].forEach((variant) => {
```

Comment:
Just simplified the tests, and changed the classes we want to test out. For example in Material You, we don't have a combination of classes with multiple props, like `textPrimary`, `outlinedPrimary` etc. This is why we need less tests.